luarules/gadgets/unit_gunship_strafe_fix.lua

```diff
@@ -0,0 +1,100 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+	return {
+		name = "Gunship Strafe Fix",
+		desc = "When gunships stop attacking their target unit with no followup command, forces them to stop strafing and enter idle behavior",
+		author = "JRTaylord: https://github.com/JRTaylord",
+		date = "May 7, 2025",
+		license = "GNU GPL, v2 or later",
+		layer = 0, -- Todo: ask which layer should this be on
+		enabled = true,
+	}
+end
+
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+local spGetUnitWeaponTarget = Spring.GetUnitWeaponTarget
+local spGetUnitCommands = Spring.GetUnitCommands
+local spGetAllUnits = Spring.GetAllUnits
+local spGetUnitDefID = Spring.GetUnitDefID
+local spEcho = Spring.Echo
+
+-- Constants
+-- Todo figure out options to make the command silent
+local CMD_OPTIONS = CMD.OPT_INTERNAL
+
+-- Gadget variables
+local gunshipDefs = {}
+local gunshipsToTrack = {}
+
+-- Only run in synced code
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+-- Helper functions
+local function isTargettingUnit(unitID, unitDefID)
+	local weaponCount = gunshipDefs[unitDefID].weaponCount
+	for weaponNum = 1, weaponCount do
+		local targetType, _, targetID = spGetUnitWeaponTarget(unitID, weaponNum)
+		if targetType == 1 and targetID then
+			return true
+		end
+	end
+	return false
+end
+
+local function stopGunshipIfNoCmdAndTarget(unitId, unitDefID)
+	local numCommands = #spGetUnitCommands(unitID, 1)
+	local hasTarget = isTargettingUnit(unitID, unitDefID)
+	if numCommands == 0 and not hasTarget then
+		spGiveOrderToUnit(unitID, CMD.STOP, {}, CMD_OPTIONS)
```

Comment:
A full stop command has a bunch of side effects, perhaps consider resetting just targeting (via `Spring.SetUnitTarget(unitID, nil)` or movement.

Reply:
I'm looking into this with controlling movement or setting the targeting. A drop in fix with a call to those methods doesn't work from my testing but I'll try a more sophisticated solution